### ISSUE TITLE
SSO Fails when local_user is enabled

### ISSUE THREAD

**Issue by sover02** [gravitational/teleport#3096]
Have a question? Please use [Our Forum](https://community.gravitational.com)

**What happened**:
Single signon redirects back to login page after SSO login when local_auth is `false`, with no error message and no logs

**What you expected to happen**:
Single signon should log user in

**How to reproduce it (as minimally and precisely as possible)**:
```
auth_service:
  authentication:
    second_factor: otp
    local_auth: false
```

```
teleport start -d
#Log into proxy webui on :3080 using SSO, it redirects back to the login page after authenticating with SSO provider
```

**Environment**:
- Teleport version (use `teleport version`): Teleport Enterprise v4.1.0git:v4.1.0-0-g3078799c go1.12.1b4

- Tsh version (use `tsh version`): N/A
- OS (e.g. from /etc/os-release):
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
```

**Browser environment**

- Browser Version (for UI-related issues): `Chrome Version 77.0.3865.120 (Official Build) (64-bit)`
- Install tools: `Used .deb package: teleport-ent_4.1.0-fips_amd64.deb`
- Others: `Behavior occurs with and without --fips`

**Relevant Debug Logs If Applicable**
- tsh --debug: `N/A`
- teleport --debug: `None that seem relevant`

**Comment by sover02** [comment]
teleport.yaml
```
teleport:
  auth_token: <key>
  auth_servers:
    - localhost:3025

  advertise_ip: <public_ip>
  nodename: <instance_id>

  storage:
    # By default teleport uses the `data_dir` directory on a local filesystem
    #type: dir
    type: dynamodb
    region: <region>
    # Name of the DynamoDB table. If it does not exist, Teleport will create it.
    table_name: teleport_example

    # Array of locations where the audit log events will be stored. by
    # default they are stored in `/var/lib/teleport/log`
    audit_events_uri: ['file:///var/lib/teleport/log', 'dynamodb://example_events']

    # Use this setting to configure teleport to store the recorded sessions in
    # an AWS S3 bucket. see "Using Amazon S3" chapter for more information.
    audit_sessions_uri: 's3://example-sessions'

auth_service:
  # enable the auth service:
  enabled: true
  # required to run multiple auth servers for HA
  cluster_name: example

  tokens:
  # this static token is used for other nodes to join this Teleport cluster
  - proxy,node:<key>
  # this token is used to establish trust with other Teleport clusters
  - trusted_cluster:<key>

  # by default, local authentication will be used with 2FA
  authentication:
    second_factor: otp
#    local_auth: false

ssh_service:
  enabled: true
  labels:
    role: teleport-auth

proxy_service:
  enabled: false
```

**Comment by sover02** [comment]
okta_connector resource:
```
kind: saml
metadata:
  name: OktaSAML
spec:
  acs: <sso acs>
  attributes_to_roles:
  - name: groups
    roles:
    - role1
    - admin
    value: okta_group1
  - name: groups
    roles:
    - role2
    value: okta_group2
  audience: <sso acs>
  cert: ""
  display: Login with Okta SSO
  entity_descriptor: |
    <xml entity descriptor info>
  sso: <sso_url>
version: v2
```

**Comment by sover02** [comment]
After a bit more testing:
`tsh` still works even though the login UI is giving us the redirect problem.

**Comment by webvictim** [comment]
This will be fixed in Teleport 4.1.3 (coming soon)
